### PR TITLE
rad weaponry update: xray nerf, bouncer egun addition

### PR DESCRIPTION
**Pull request by 00ze-cyclone**
# Document the changes in your pull request

Changes things with our energy radioactive weapons to make them more funky and less egun and lasergun+.

## Xray gun:
Just brutally nerf it because 20 lasers that go through walls AND give enough rads to force you to get medicine for it for the meager tradeoff of 5 less burn was stupid. Now xray lasers deal 10 burn only and the xray laser only has 14 shots, it's still VERY good, just not superior to the laser gun in literally every way. It also got moved from basic beam weaponry to radiactive weaponry because it's a gun shooting radiation why the fuck wasn't it here?

## Bouncer
Since the xray gun left the beam weaponry node welcome the new beam weapon going there so the temp gun doesn't feel lonely: **The Bouncer Energy Gun**
All the projectiles of the bouncer egun will ricochet on ANYTHING 5 times before dissipating.
It has 2 firemodes:
-disable: 16 bouncing disabler ball, deal 20 stamina
-suffocate: 16 bouncing anoxia balls, deal 6 oxygen damage

I made it have a suffocate mode instead of kill because that gun is pretty much guaranteed to hit random bystanters so it only deal damages that go away by themselves eventually, also because I didn't wanted the bouncer to be just egun but it ricochet.

Also fixed the charge overlay of the mindflayer because it was borked.

# Spriting
Thank you Papi

Bouncer on disable mode
bouncer cyan
bouncer disable sprite

Bouncer on suffocate mode
bouncer purple
bouncer suffocate sprite

# Wiki Documentation

Xray gun:
-14 xray lasers: 10 burn, 300 rads (unchanged)

Bouncer egun:
Unlocked by the basic beam weaponry node, printed at sec techfab, cost:
-4000 iron
-10000 glass
-1000 silver
-1200 titanium
-20u of sorium

All projectiles bounce on everything and up to 5 times:
-16 bouncing disabler balls: 20 stamina
-16 bouncing anoxia balls: 6 oxygen

# Changelog

:cl:  00ze-cyclone and Papi
rscadd: Add the bouncer energy gun 
bugfix: fixed the charge overlay of the mind flayer
tweak: reduced ammo of xray to 14 beams, xray lasers deal 10 burn and 300 rads now
imageadd: added bouncer egun sprite made by Papi
/:cl:

### CODE REVIEW

**Comment by 00ze-cyclone** [comment]
Let's make it functional actually before PRing

**Comment by 00ze-cyclone** [comment]
Eh fuck it I'll use the bot here to help do the fixing.

**Comment by UltraBigRat** [comment]
finally a buff to ipc/plasmeme securitae

**Comment by Marmio64** [comment]
dont see why advanced energy gun needs to be moved from it's niche to practically being a ballistic gun that shoots lasers which is also only operable if you're either a plasmaman, wearing a radsuit, or have radmeds with you, especially since you now have to make weaponkit crap and apply it to an already existing gun to get it (it's no longer free). If the guns too strong nerf the capacity so it still fits within it's niche rather than no longer being wanted by people who didn't choose plasmaman as their race.

**Comment by 00ze-cyclone** [comment]
> dont see why advanced energy gun needs to be moved from it's niche to practically being a ballistic gun that shoots lasers which is also only operable if you're either a plasmaman, wearing a radsuit, or have radmeds with you, especially since you now have to make weaponkit crap and apply it to an already existing gun to get it (it's no longer free). If the guns too strong nerf the capacity so it still fits within it's niche rather than no longer being wanted by people who didn't choose plasmaman as their race.

gunkits aren't merged yet and might actually not be.
Aegun doesn't have a niche right now, it's egun+.
New egun you can still use it as a human without any radmeds, sure you'll die of cancer after but you can still use that reloadable gun with 4 firemodes to kill your opponent before you die, then you get recovered by your buddies and get brought back. It's a sacrifice in exchange of a tactical advantage.

**Comment by TheRyeGuyWhoWillNowDie** [comment]
man if only SOMEONE had thought of some way to limit security's powergame ability and make the armory not worthless (gun kits)

**Comment by TheRyeGuyWhoWillNowDie** [comment]
also im an aeg hater as much as the next guy, but having to hold still for like 5 seconds to insert a limited material into a gun to recharge it by like 2 lasers seems really ridiculously bad

**Comment by Marmio64** [comment]
> > dont see why advanced energy gun needs to be moved from it's niche to practically being a ballistic gun that shoots lasers which is also only operable if you're either a plasmaman, wearing a radsuit, or have radmeds with you, especially since you now have to make weaponkit crap and apply it to an already existing gun to get it (it's no longer free). If the guns too strong nerf the capacity so it still fits within it's niche rather than no longer being wanted by people who didn't choose plasmaman as their race.
> 
> gunkits aren't merged yet and might actually not be. Aegun doesn't have a niche right now, it's egun+. New egun you can still use it as a human without any radmeds, sure you'll die of cancer after but you can still use that reloadable gun with 4 firemodes to kill your opponent before you die, then you get recovered by your buddies and get brought back. It's a sacrifice in exchange of a tactical advantage.

It's niche is supposed to be that it self recharges but doesn't have the capacity of a lasgun or disabler, which while it is a better egun, it's not strictly superior to the other options. This is giving it a new niche, but a really bad one because the ability to slowly reload a gun in exchange for a crap ton of rads isn't very good unless maybe you're a ipc/plasmaman.

**Comment by 00ze-cyclone** [comment]
> also im an aeg hater as much as the next guy, but having to hold still for like 5 seconds to insert a limited material into a gun to recharge it by like 2 lasers seems really ridiculously bad

Only one second per sheet, in 5 second it’s fully refilled like if you had put it in a T4 recharger but without needing a T4 recharger

**Comment by 00ze-cyclone** [comment]
> > > dont see why advanced energy gun needs to be moved from it's niche to practically being a ballistic gun that shoots lasers which is also only operable if you're either a plasmaman, wearing a radsuit, or have radmeds with you, especially since you now have to make weaponkit crap and apply it to an already existing gun to get it (it's no longer free). If the guns too strong nerf the capacity so it still fits within it's niche rather than no longer being wanted by people who didn't choose plasmaman as their race.
> > 
> > 
> > gunkits aren't merged yet and might actually not be. Aegun doesn't have a niche right now, it's egun+. New egun you can still use it as a human without any radmeds, sure you'll die of cancer after but you can still use that reloadable gun with 4 firemodes to kill your opponent before you die, then you get recovered by your buddies and get brought back. It's a sacrifice in exchange of a tactical advantage.
> 
> It's niche is supposed to be that it self recharges but doesn't have the capacity of a lasgun or disabler, which while it is a better egun, it's not strictly superior to the other options. This is giving it a new niche, but a really bad one because the ability to slowly reload a gun in exchange for a crap ton of rads isn't very good unless maybe you're a ipc/plasmaman.

Only gun with a higher capacity is the laser, aegun has 20 disable beams like the disabler. It is completely superior to egun, it isn’t to laser gun because laser is better than any egun if you just want to shoot lasers.

**Comment by cuackles** [comment]
the sprite is ok it does look like the standart e gun but with enough diferences to be noticable at a quick glance, i do have a small issue with this part here that seems a bit muddy and there is a big glob with the same color, i would like to see if you are able to change that to be more clear, apart from that i would like to know if you are able to post a video of the bouncer shoting cus i would like to see the proyectile in game
image
thats the part i would like to see if you are able to make more clear

File: code/modules/projectiles/guns/energy/energy_gun.dm [hunk 1]
```diff
@@ -132,58 +134,44 @@
...
+			return
+		to_chat(user, span_notice("You start delicately inserting [I] in [src]'s reactor, refueling it."))
+		if(do_after(user, 1 SECONDS, src))
+			I.use(1)
+			cell.give(100*charge_multiplier)
+			user.radiation += (60*charge_multiplier) //You are putting you hand into a nuclear reactor to put more uranium in it
+		else
+			if(!(previous_loc == user.loc))
+				to_chat(user, span_boldwarning("You move, bumping your hand on [src]'s nulear reactor's core!")) //when I said devoid of ANY safety measures I meant it
+				user.adjustToxLoss(5*charge_multiplier) //straigth toxin damage rather than rads because we want the user to be punished immediately for moving, not in 2 hours
```
Comment (Moltijoe) [inline code/modules/projectiles/guns/energy/energy_gun.dm]: if it's "radiation tox", having rad immunity should prevent this

Comment (Sniblet) [reply code/modules/projectiles/guns/energy/energy_gun.dm]: nulear reactor's

**Comment by Sniblet** [comment]
do anoxia rays break IPCs

**Comment by Hislyna** [comment]
The Bouncer sounds like it would be more of an un/intentional grief weapon, sounds fun but in practise i think itll be a useless bloat weapon. And hey at least we have race specific weapons for sec now!

**Comment by 00ze-cyclone** [comment]
> do anoxia rays break IPCs

Even if it somehow did oxygen to IPC there's #15990

**Comment by 00ze-cyclone** [comment]
> The Bouncer sounds like it would be more of an un/intentional grief weapon, sounds fun but in practise i think itll be a useless bloat weapon. And hey at least we have race specific weapons for sec now!

In maint it would be pretty effective since you could make the projectiles ricochet at an angle to cover the narrow rooms almost fully. It's also way less griefy than tesla revolver since you won't do long lasting damage.

File: code/modules/projectiles/guns/energy/energy_gun.dm (hunk 1 above)
Comment (00ze-cyclone) [reply code/modules/projectiles/guns/energy/energy_gun.dm]: Nah it's just mega toxicity because that gun was built with any safety or sanity

**Comment by Hislyna** [comment]
> > The Bouncer sounds like it would be more of an un/intentional grief weapon, sounds fun but in practise i think itll be a useless bloat weapon. And hey at least we have race specific weapons for sec now!
> 
> In maint it would be pretty effective since you could make the projectiles ricochet at an angle to cover the narrow rooms almost fully. It's also way less griefy than tesla revolver since you won't do long lasting damage.

It'll be interesting to see it test merged to say the least

**Comment by ChuChu42** [comment]
i enjoy all 3 of theese additions and just because they arent straight nerfs but more of reworks

also i tought you gave the bar bouncer a egun after reading the title

**Comment by 00ze-cyclone** [comment]
> i enjoy all 3 of theese additions and just because they arent straight nerfs but more of reworks
> 
> also i tought you gave the bar bouncer a egun after reading the title

well the xray is a straight nerf tbf, but yeah I didn't wanted to just nerf the aegun since my main gripe with it was more of how not that interesting the concept of the gun was.

**Comment by MajManatee** [comment]
AEgun changes are terrible. the rest is cool. Let it self charge.

**Comment by 00ze-cyclone** [comment]
> AEgun changes are terrible. the rest is cool. Let it self charge.

How about less rads, like only 400 per full reload, or maybe 300? Because I am NOT letting the egun+ exist, egun but you don't even need to care about reloading is just lame.

**Comment by MajManatee** [comment]
You expect anyone to use eguns if they need to be reloaded, and doing so gets you radioactive after 2 reloads?

**Comment by 00ze-cyclone** [comment]
> You expect anyone to use eguns if they need to be reloaded, and doing so gets you radioactive after 2 reloads?

Yes, 20 lasers is good enough to solve a lot of problems, you get regular lasers and xray in just one gun so you can shoot someone with lasers then xray once they take cover, the anoxia is energy instead of laser so it's basically AP laser and only 50 oxy damage are needed to crit so it's akin to 40 AP beams that deal 16 damage, if you are willing to deal with cancer afterward you have a lot of firepower.

**Comment by JamieD1** [comment]
Atomise this, so many different changes is stupid.

**Comment by 00ze-cyclone** [comment]
Aegun change is removed from the PR, I'll return to it later

**Review by MajManatee — CHANGES_REQUESTED** [review]
more atomic

**Review by KillerOrcaCora — CHANGES_REQUESTED** [review]
Banding a bit and could use better shading.

**Review by JamieD1 — APPROVED** [review]

**Comment by JamieD1** [comment]
When orca is happy, can merge this.

**Review by KillerOrcaCora — APPROVED** [review]

**Comment by MajManatee** [comment]
@00ze-cyclone DMI conflicts

**Review by MajManatee — APPROVED** [review]
finally

**Pull request merged**